apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-control.js

```diff
@@ -1,30 +1,81 @@
 /**
  * External dependencies
  */
-import { isEmpty } from 'lodash';
+import { isEmpty, noop } from 'lodash';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
 import { withInstanceId } from '@wordpress/compose';
 import { BaseControl } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { parse as parseBlocks } from '@wordpress/blocks';
+import { BlockPreview } from '@wordpress/block-editor';
+
+/**
+ * It renders the block preview content for the template.
+ * It the templates blocks are not ready yet or not exist,
+ * it tries to render a static image, or simply return null.
+ */
+class TemplateSelectorItem extends Component {
+	state = {
+		blocks: [],
+	};
+
+	constructor( props ) {
+		super();
+
+		if ( props.rawBlocks ) {
+			this.state.blocks = parseBlocks( props.rawBlocks );
```

Comment:
I think this is probably the same thing as `const blocks = useMemo( () => parseBlocks( props.rawBlocks ), [ props.rawBlocks ] )` if you ever want to try a functional component. Not that important though.

Reply:
https://github.com/Automattic/wp-calypso/pull/35022/commits/59beaa824bf0d9e04b83921089896414a482d5f5